pp.c

```diff
@@ -6986,10 +6994,14 @@ PP(pp_anonconst)
 {
     dSP;
     dTOPss;
-    SETs(sv_2mortal((SV *)newCONSTSUB(SvTYPE(CopSTASH(PL_curcop))==SVt_PVHV
-                                        ? CopSTASH(PL_curcop)
-                                        : NULL,
-                                      NULL, SvREFCNT_inc_simple_NN(sv))));
+
+    CV* constsub = newCONSTSUB(
+        SvTYPE(CopSTASH(PL_curcop))==SVt_PVHV ? CopSTASH(PL_curcop) : NULL,
+        NULL,
+        SvREFCNT_inc_simple_NN(sv)
+    );
+
+    SETs(refto( sv_2mortal((SV *)constsub) ));
```

Comment:
Whitespacing here seems to be off. Remove the extra whitespace inside the parens.

Reply:
> The point of using the flag on `OP_ANONCODE` was to shield them from that - the same surely should apply to `OP_ANONCONST`.

As @atoomic pointed out, though, OP_ANONCONST relates to an experimental feature. Given that, would it be more advantageous to keep `pp_anonconst` a bit simpler, and always do the ref?

Reply:
> Whitespacing here seems to be off. Remove the extra whitespace inside the parens.

OK. But for my own reference, is there a style guide that dictates spacing? I personally find it a bit easier to grok with the spaces, FWIW.

Reply:
(Ugh, hate that the email versions of these comments tend to quote extra lines, making it appear to be a conversation about the `newCONSTSUB()` call rather than `SETs()`.)

FWIW I have no objection to the extra spaces here - they're correctly balanced, and make the line more readable by giving the eye extra reference points to help it see how to parse the code. In the same way in perl code I'd tend to write `$array[$index]` but `$array[ $subarray[$subindex] ]`.

It won't kill me if they are avoided in any specific case - it's really a judgement call when they improve things and when they detract - but I'd hate to see a style guide mandating against them.

Reply:
> As @atoomic pointed out, though, OP_ANONCONST relates to an experimental feature. Given that, would it be more advantageous to keep `pp_anonconst` a bit simpler, and always do the ref?

Ah, hrm. Somewhat complicated question. On the one hand perhaps it is still experimental, but on the other hand it's been around quite a while now:

```
$ git blame regen/opcodes | grep anonconst
b77472f98ff (Father Chrysostomos      2015-01-18 16:37:03 -0800 576) anonconst  anonymous constant      ck_null         ds1
```

The commit message says it's to do with the `:const` sub attribute, according to perlexperiment.pod introduced in 5.22. Which feels like quite a long time for a running experiment to have been sat for... Hrm.

Reply:
CPAN, at least, doesn't seem to show appreciable usage of the anonconst op.

https://grep.metacpan.org/search?size=20&q=anonconst&qd=&qft=&qci=on

Reply:
Per IRC, I’ve resolved this by adding the OPf_REF, and I _believe_ there’s consensus that the spacing is good as-is.